File: core/domain/document.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Optional


@dataclass
class Document:
    id: str
    content_hash: str
    created_at: str
    updated_at: str
    title: Optional[str] = None
    date_month: Optional[str] = None
    png_path: Optional[str] = None
    duplicate_of: Optional[str] = None
    additional_context: str = ""
    linked_contexts: list = field(default_factory=list)  # list of UUID strings

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "content_hash": self.content_hash,
            "created_at": self.created_at,
            "updated_at": self.updated_at,
            "title": self.title,
            "date_month": self.date_month,
            "png_path": self.png_path,
            "duplicate_of": self.duplicate_of,
            "additional_context": self.additional_context,
            "linked_contexts": self.linked_contexts,
        }

File: server/core/model/document.go
package model

import "time"

type FileType string

const (
	FileTypePNG  FileType = "png"
	FileTypeJPG  FileType = "jpg"
	FileTypeJPEG FileType = "jpeg"
	FileTypeTXT  FileType = "txt"
	FileTypeMD   FileType = "md"
	FileTypeWEBM FileType = "webm"
	FileTypeWAV  FileType = "wav"
	FileTypeMP3  FileType = "mp3"
	FileTypeM4A  FileType = "m4a"
	FileTypeOGG  FileType = "ogg"
	FileTypeFLAC FileType = "flac"
	FileTypeMP4  FileType = "mp4"
)

// IsImage reports whether the file type is a still image.
func (f FileType) IsImage() bool {
	return f == FileTypePNG || f == FileTypeJPG || f == FileTypeJPEG
}

// IsTranscribable reports whether the file type is media that the transcribe
// stage knows how to handle (audio formats and MP4 video).
func (f FileType) IsTranscribable() bool {
	switch f {
	case FileTypeWEBM, FileTypeWAV, FileTypeMP3, FileTypeM4A, FileTypeOGG, FileTypeFLAC, FileTypeMP4:
		return true
	}
	return false
}

// IsText reports whether the file type is plain text or markdown.
func (f FileType) IsText() bool { return f == FileTypeTXT || f == FileTypeMD }

type Document struct {
	ID          string
	ContentHash string
	CreatedAt   time.Time
	UpdatedAt   time.Time
	Title       *string
	DateMonth   *string
	// MediaPath is the vault path to the source media file (image, audio, etc.).
	// Stages branch on the document's file type to decide how to consume it.
	MediaPath         *string
	DuplicateOf       *string
	AdditionalContext string
	LinkedContexts    []string // context entry IDs
	Series            *string
}
